### ISSUE TITLE
Crash when deleting / updating preset

### ISSUE THREAD

**Issue by x42** [drobilla/lilv#32]
Recent lv2kit, lilv v0.24.6-2-gd1c9d1b and Ardour (6.0-pre0-3173-g563a8b15e0)

Steps to reproduce: Create a LV2 preset, change a parameter, save the preset.

Expected behavior: Old preset is deleted, a new one (with same name) is created. Known to work with 0.24.2.

```gdb
Thread 1 "ardour-6.0.pre0" received signal SIGSEGV, Segmentation fault.
0x00007ffff2db32aa in zix_tree_begin (t=0x0) at ../libs/lv2kit/libs/lilv/src/zix/tree.c:622
622		if (!t->root) {
(gdb) bt
#0  0x00007ffff2db32aa in zix_tree_begin (t=0x0) at ../libs/lv2kit/libs/lilv/src/zix/tree.c:622
#1  0x00007ffff2dacc14 in lilv_state_delete (world=0x555556e26840, state=0x55555720ac10) at ../libs/lv2kit/libs/lilv/src/state.c:1257
#2  0x00007ffff7c6503f in ARDOUR::LV2Plugin::do_remove_preset(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >) (this=0x55555c045650, name="2048")
    at ../libs/ardour/lv2_plugin.cc:1712
#3  0x00007ffff79cd2f9 in ARDOUR::Plugin::remove_preset(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >) (this=0x55555c045650, name="2048")
    at ../libs/ardour/plugin.cc:140
#4  0x0000555556867bfc in PlugUIBase::save_plugin_setting() (this=0x55555caf87c0) at ../gtk2_ardour/plugin_ui.cc:671
```

**Comment by x42** [comment]
PS. despite the crash the preset is deleted (at least it's no longer present on disk).

**Comment by drobilla** [comment]
Should be fixed in 1bb1de7, thanks.

I still don't feel great about this code at all, but I fixed several other things along the way, so some manual testing with Ardour of various cases would be a good idea.
